homeassistant/components/weather/darksky.py

```diff
@@ -0,0 +1,189 @@
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_UNITS): vol.In(['auto', 'si', 'us', 'ca', 'uk', 'uk2']),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=3)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Dark Sky weather."""
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    name = config.get(CONF_NAME)
+
+    units = config.get(CONF_UNITS)
+    if not units:
+        units = 'si' if hass.config.units.is_metric else 'us'
+
+    dark_sky = DarkSkyData(
+        config.get(CONF_API_KEY), latitude, longitude, units)
+
+    add_devices([DarkSkyWeather(name, dark_sky)], True)
+
+
+class DarkSkyWeather(WeatherEntity):
+    """Representation of a weather condition."""
+
+    def __init__(self, name, dark_sky):
+        """Initialize Dark Sky weather."""
+        self._name = name
+        self._dark_sky = dark_sky
+
+        self._ds_data = None
+        self._ds_currently = None
+        self._ds_hourly = None
+        self._ds_daily = None
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return ATTRIBUTION
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def temperature(self):
+        """Return the temperature."""
+        return self._ds_currently.get('temperature')
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_FAHRENHEIT if 'us' in self._dark_sky.units \
+            else TEMP_CELSIUS
+
+    @property
+    def humidity(self):
+        """Return the humidity."""
+        return self._ds_currently.get('humidity') * 100.0
+
+    @property
+    def wind_speed(self):
+        """Return the wind speed."""
+        return self._ds_currently.get('windSpeed')
+
+    @property
+    def pressure(self):
+        """Return the pressure."""
+        return self._ds_currently.get('pressure')
+
+    @property
+    def condition(self):
+        """Return the weather condition."""
+        return self._ds_currently.get('summary')
+
+    @property
+    def forecast(self):
+        """Return the forecast array."""
+        return [{
+            ATTR_FORECAST_TIME:
+                datetime.fromtimestamp(entry.d.get('time')).isoformat(),
+            ATTR_FORECAST_TEMP: entry.d.get('temperature')}
+                for entry in self._ds_hourly.data]
+
+    @property
+    def hourly_forecast_summary(self):
+        """Return a summary of the hourly forecast."""
+        return self._ds_hourly.summary
+
+    @property
+    def daily_forecast_summary(self):
+        """Return a summary of the daily forecast."""
+        return self._ds_daily.summary
+
+    @property
+    def state_attributes(self):
```

Comment:
You should not overwrite this but implement `device_state_attributes`. Those will update the state attributes in the entity base class.